### PR TITLE
Docs: Add dependency update guidance to repository

### LINKED ISSUE #296: Dependency Update Guidance
As a project that augments Supply Chain Security (Governance), Lula should have documented guidance for how the project performs responsible use of dependency management tools such as Renovate. A primary goal here ensuring that the process is understood as well as the benefits that present themselves by performing dependency updates with additional scrutiny. 

## Required
- Evaluation of code changes between versions
- Checksum / Hash validation

## Optional
- Approval messages that attest to the required actions occurring

### PR DESCRIPTION
**Pull request by brandtkeller**
Closes #296 

Adds dependency update guidance around the use of Renovate for keeping project dependencies up-to-date. 

Open to feedback on redundant or missing items.

### CODE REVIEW

File: docs/dependency-updates.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - NPM integrity
+    - tarball integrity validation
+  - Workflow Integrity
+    - Tag Commit checksum
+- Annotation of the reviewed dependency updates for approval
+  - Include any relevant notes or considerations
+  - Include steps to validate the dependency updates
+
+### Notes
+- Validation of the checksums is currently a manual process an a byproduct of not yet capturing the provenance of Renovates checksum process. Given that no single version of Renovate is being used (this is the non-self-hostedGitHub application), we do not track updates to the renovate runtime itself.
```
Comment (CloudBeard) [inline docs/dependency-updates.md]: Should it be as not an for "manual process an a byproduct " in line 32?

Nothing from me to add, the doc covered everything I thought of for dependency updates.

**Review by mike-winberry — DISMISSED** [review]
Looks good, do we want to throw example links in there?

**Comment by brandtkeller** [comment]
> Looks good, do we want to throw example links in there?

@mike-winberry Open to thoughts here! Are you thinking some instance of an update and the process involved as an example?

**Comment by mike-winberry** [comment]
> > Looks good, do we want to throw example links in there?
> 
> @mike-winberry Open to thoughts here! Are you thinking some instance of an update and the process involved as an example?

 I was thinking some like example commands or preferred tooling for the `- Validate the integrity / provenance of the updated dependency` portion.

**Review by CloudBeard — APPROVED** [review]

**Review by mike-winberry — APPROVED** [review]

**Pull request merged**